### ISSUE TITLE
Key Input

### ISSUE THREAD

**Issue by communityus** [l3lessed/l3lessed-DFUnity-Mods#2]
On testing in 10.26:

I can change the block key to something like: F

That works but then Swing Weapon only swings shield I think as if to Perry?

It looks like it may just be because all of these haven't been implemented yet:
```
        private bool parryKeyPressed;
        private bool attackKeyPressed;
        private bool offHandKeyPressed;
        private bool handKeyPressed;
        private bool mainHandKeyPressed;
```

**Comment by l3lessed** [comment]
Sorry, the block key is also the offhand key.

The setup I use is left mouse for main attack, right mouse for offhand attack, and E for activate.

However, just discovered a shield bug when using left handiness. setting.

**Comment by communityus** [comment]
Yeah I see the bug with left handiness. I assume this is based on "click to attack" being toggled as well?

Perhaps on some weapons work? I've been trying with the fist and the starting iron saber as a wood elf female and all I can do is move the shield no matter the settings I try I can't swing my sword. I am not testing in combat with enemy just swinging in air - not sure if I need to be fighting an enemy? I'll keep playing with it. I know it is early stages and a lot of new work here. just wanted to report it in case others find it to.

**Comment by l3lessed** [comment]
Odd, I'm not running into this issue if there is no left handiness setting checked. Also, I'm not hearing about this issue on the forums from other players,

I did forget to mention, mouse drag attacking does not work at all right now. I have to rebuild that code so it can work with both hands.

Any other mods installed?

Also, doesn't seem like a keybind issue, but ensure your are using the proper keycode property name from this list: https://docs.unity3d.com/ScriptReference/KeyCode.html

**Comment by l3lessed** [comment]
Also, all those bools are old and unused. I was using them for a bug riddled, simple bool key input check routine I made on the fly. They are not required anymore, now that key inputs are managed by a proper timed que system.
